models/articles.models.js

```diff
@@ -14,14 +14,27 @@ exports.selectArticleById = (article_id) => {
   });
 };
 
-exports.selectArticles = () => {
-  let queryString =
-    "SELECT articles.author,articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;";
-  return db.query(queryString).then(({ rows }) => {
-    return rows;
-  });
-};
-
+exports.selectArticles = (topic) => {
+if (!topic) {
+     let queryString =
+     "SELECT articles.author,articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;";
```

Comment:
You should be able to conditionally  build up the query string (if there is a topic) and then only have one query to the database. Currently this is not very DRY.